architecture-building-systems/CityEnergyAnalyst-GUI
# 2536 technology editor

This PR implements the functions added in https://github.com/architecture-building-systems/CityEnergyAnalyst/pull/2554

New Functions
-
- Able to edit and export databases in the new database editor
- Will provide basic validation for cells

Testing
-
- Go through workflow of creating new scenario and select `create own database` option
- Should be brought to database editor and error should show `could not find/verify database`
  - Should also show error when opening a scenario with invalid database structure e.g. rename one of the excel files
- Assign a database using the script
- Select one database and enter an invalid value into a cell
- A error message should show where the error is found e.g. sheet, row, col
- Users will not be able to select another database/save database if the error is not corrected
- Adding rows and undo/redo should still show the correct error information
- Correct all errors and save the database
- There will be options to go to `Archetype Mapper` and to `Export Database`
- Try and export database and check if it appears in the desired path

Build: https://we.tl/t-xv5xZkHVSK
